src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -81,21 +84,32 @@ public static LoginData performAutoLogin(StartupParams params) {
         loginData.setCredentials(username, password);
 
         try {
-            System.out.println("Auto logging in (1/2)");
-            usernameLogin(loginData);
+            if(isSidAutoLogin) loginData.setSid(sid, server + ".darkorbit.com");
+            else {
+                System.out.println("Auto logging in (1/2)");
+                usernameLogin(loginData);
+            }
             System.out.println("Loading spacemap (2/2)");
             findPreloader(loginData);
         } catch (IOException e) {
             System.err.println("IOException trying to perform auto login, servers may be down");
             e.printStackTrace();
         } catch (WrongCredentialsException e) {
+            if(isSidAutoLogin) {
+                System.err.println("Wrong SID");
```

Comment:
```suggestion
                System.err.println("Expired SID in login properties file, attempting re-connect with user & pass");
```
descriptive errors will help users when they find themselves fighting this, which will eventually occur

Reply:
On this line we can not grantee that attempt of re-login will happens, for example login/password is absent

Reply:
That's true, we cannot guarantee it'll happen
however, we just saying we're re-attempting, the next error you'll get after this, will be the one that is like "For auto login you need either user & pass, or sid & server", so you will be able to see what happened